### PR TITLE
Drop GHA pr-commands and add GHA Rituals. Use pak to install

### PR DESCRIPTION
**Pull request by wch**
This PR is for testing modifications to #3226.

### CODE REVIEW

**Comment by wch** [comment]
I've fixed the Mac build issue with textshaping, by installing some system dependencies.

Windows build of bslib still fails. I believe it's because of filenames that are too long. Here's one example:

https://github.com/rstudio/shiny/runs/1597854296?check_suite_focus=true#step:10:376

The file is `rstudio-bslib-338b05f/inst/lib/bootstrap-accessibility-plugin/plugins/js/bootstrap-accessibility_1.0.3.min.js`, inside of `C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\Rtmp4K8oTQ\\file10406c64986/src/contrib/bslib_0.2.3.9000_338b05fc5aa551c0cb62dae2ce73ad8c2ceb8403.tar.gz-tree`. The total length the combined path is is 263 characters, which exceeds the max length of a file on a typical Windows installation.

It would probably be good for pak to use a shorter directory name, but it might also be a good for bslib to also use shorter path.

**Comment by wch** [comment]
I tried using `url::https://github.com/rstudio/bslib/archive/master.tar.gz` for bslib, to try to work around the filename length issue. However, pak apparently does not understand `url::` remotes, so installation fails. https://github.com/rstudio/shiny/pull/3230/checks?check_run_id=1598093331#step:6:26

https://github.com/r-lib/pak/issues/253

**Comment by wch** [comment]
I added a workaround for the Windows filename length issue, but now it's failing due to https://github.com/r-lib/pak/issues/254

**Comment by wch** [comment]
r-lib/pak#254 has been fixed, and it looks like it finally all works, after some more tweaks to the Windows path stuff.

File: .github/workflows/rituals.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      - name: Mac systemdeps
+        if: runner.os == 'macOS'
+        run: |
+          brew cask install xquartz
+          brew install cairo
+
+      - name: Install dependencies
+        shell: Rscript {0}
+        run: |
+          pak::pkg_install("curl") # install before installing deps
```
Comment (schloerke) [inline .github/workflows/rituals.yaml:80]: Was nice for `remotes`. Not needed for `pak`.
```suggestion
```

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -45,49 +45,65 @@ jobs:
...
+        if: runner.os == 'Windows'
+        run: |
+          New-Item -Path "C:\" -Name "tmp" -ItemType Directory
+          echo "TMPDIR=c:\tmp" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
 
       - name: Install dependencies
         run: |
-          remotes::install_deps(dependencies = TRUE)
-          remotes::install_cran("rcmdcheck")
+          pak::pkg_install("curl") # install before installing deps
```
Comment (schloerke) [inline .github/workflows/R-CMD-check.yaml:92]: Was needed for `remotes`. Not needed for `pak`.
```suggestion
```

File: .github/workflows/rituals.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        shell: Rscript {0}
+        run: |
+          pak::local_system_requirements(execute = TRUE)
+
+      # Install Cairo system dependencies
+      - name: Mac systemdeps
+        if: runner.os == 'macOS'
+        run: |
+          brew cask install xquartz
+          brew install cairo
```
Comment (schloerke) [inline .github/workflows/rituals.yaml:75]: Only going to leave linux based instructions

```suggestion
```

File: .github/workflows/rituals.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+      - name: Cache R packages
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ${{ matrix.config.os }}-${{ steps.install-r.outputs.installed-r-version }}-1-${{ hashFiles('.github/r-depends.rds') }}
+          restore-keys: ${{ matrix.config.os }}-${{ steps.install-r.outputs.installed-r-version }}-1-
+
+      - name: Install system dependencies
+        if: runner.os == 'Linux'
```
Comment (schloerke) [inline .github/workflows/rituals.yaml:65]:
```suggestion
        # if: runner.os == 'Linux'
```

File: .github/workflows/rituals.yaml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        run: |
+          brew cask install xquartz
+          brew install cairo
+
+      - name: Install dependencies
+        shell: Rscript {0}
+        run: |
+          pak::pkg_install("curl") # install before installing deps
+          pak::local_install_dev_deps(upgrade = TRUE)
+          pak::pkg_install("rcmdcheck")
```
Comment (schloerke) [inline .github/workflows/rituals.yaml:82]: Not checking in this test
```suggestion
```

**Comment by schloerke** [comment]
File names have been reduced in bslib. Still going to have to wait for https://github.com/r-lib/pak/issues/254 to be fixed

**Review by schloerke — APPROVED** [review]

**Pull request merged**